code/modules/surgery/organs/cybernetic_implants.dm

```diff
@@ -0,0 +1,216 @@
+	owner.permanent_sight_flags |= SEE_MOBS
+	owner.permanent_sight_flags |= SEE_OBJS
+	owner.permanent_sight_flags |= SEE_TURFS
+	update_eye_color("Your vision is augmented!")
+
+/obj/item/cybernetic_implant/eyes/thermals
+	name = "thermals implant"
+	desc = "These cybernetic eyes will give you Thermal vision. Vertical slit pupil included."
+	eye_color = "FC0"
+	implant_color = "#FFCC00"
+	flash_protect = -1
+
+/obj/item/cybernetic_implant/eyes/thermals/function()
+	if(!owner)
+		return
+
+	owner.sight |= SEE_MOBS
+	owner.permanent_sight_flags |= SEE_MOBS
+	update_eye_color("You see prey everywhere you look...")
+
+/obj/item/cybernetic_implant/eyes/emp_act(severity)
+	if(severity > 1)
+		if(prob(5))
+			return
+	var/save_sight = owner.sight
+	owner.sight &= 0
+	owner.disabilities |= BLIND
+	spawn(50)
+	owner.sight |= save_sight
+	owner.disabilities &= ~BLIND
+
+
+//[[[[BRAIN]]]]
+
+/obj/item/cybernetic_implant/brain
+	name = "cybernetic brain implant"
+	desc = "injectors of extra sub-routines for the brain"
+	icon_state = "brain_implant"
+
+/obj/item/cybernetic_implant/brain/New()
+	var/icon/overlay = new /icon('icons/obj/surgery.dmi',"brain_implant_overlay")
+	overlay.ColorTone(implant_color)
+	overlays |= overlay
+	..()
+
+/obj/item/cybernetic_implant/brain/emp_act(severity)
+	var/stun_amount = 5 + (severity-1 ? 0 : 5)
+	owner.Stun(stun_amount)
+	return stun_amount
+
+/obj/item/cybernetic_implant/brain/anti_drop
+	name = "anti-drop implant"
+	desc = "This cybernetic brain implant will allow you to force your hand muscles to contract, preventing item dropping. Twitch ear to toggle."
+	var/active = 0
+	var/l_hand_ignore = 0
+	var/r_hand_ignore = 0
+	implant_color = "#DE7E00"
+
+/obj/item/cybernetic_implant/brain/anti_drop/function()
+	action_button_name = "Toggle Anti-Drop"
+
+/obj/item/cybernetic_implant/brain/anti_drop/ui_action_click()
+	active = !active
+	if(active)
+		if(owner.l_hand)
+			if(NODROP in owner.l_hand.flags)
+				l_hand_ignore = 1
+			else
+				owner.l_hand.flags |= NODROP
+				l_hand_ignore = 0
+
+		if(owner.r_hand)
+			if(NODROP in owner.r_hand.flags)
+				r_hand_ignore = 1
+			else
+				owner.r_hand.flags |= NODROP
+				r_hand_ignore = 0
+
+		if(!owner.r_hand && !owner.l_hand)
+			owner << "<span class='notice'>You are not holding any items, your hands relax...</span>"
+			active = 0
+		else
+			var/noodles = 0
+			noodles += !l_hand_ignore && owner.l_hand ? 1 : 0
+			noodles += !r_hand_ignore && owner.r_hand ? 2 : 0
+			switch(noodles)
+				if(1)
+					owner << "<span class='notice'>Your left hand's grip tightens.</span>"
+				if(2)
+					owner << "<span class='notice'>Your right hand's grip tightens.</span>"
+				if(3)
+					owner << "<span class='notice'>Both of your hand's grips tighten.</span>"
+	else
+		if(!l_hand_ignore && owner.l_hand)
+			owner.l_hand.flags &= ~NODROP
+		if(!r_hand_ignore && owner.r_hand)
+			owner.r_hand.flags &= ~NODROP
+		owner << "<span class='notice'>Your hands relax...</span>"
```

Comment:
You can put NODROP clothing on (so you can pick up cursed stuff and put it on, only to be unable to take it off again), so you could use this by taking gloves in your hand, using this implant, putting the gloves on and then summoning an armblade to have a "free" armblade. I suggest you use direct references to these items (with a loc == owner check)